src/controllers/scripting/legacy/controllerscriptenginelegacy.cpp

```diff
@@ -660,7 +654,7 @@ void ControllerScriptEngineLegacy::handleScreenFrame(
         emit previewRenderedScreen(screenInfo, screenDebug);
     }
 
-    QByteArray input(std::bit_cast<const char*>(frame.constBits()), frame.sizeInBytes());
+    QByteArray input(reinterpret_cast<const char*>(frame.constBits()), frame.sizeInBytes());
```

Comment:
AFAIR, this is what I was using, but I was requested to change it to `bit_cast`. Perhaps you could keep this cast specific to Mac?

Reply:
A comment sounds good yeah!

Reply:
> Perhaps a comment that this should be refactored to a std::bit_cast once we drop support for macOS 11 (or build on a newer runner)?

is this specific to the build environment or to the runner? Cuz the runner was recently updated #13296. 

Here's a little more context for reinterpret_cast vs std::bit_cast: https://stackoverflow.com/a/53402217

Reply:
Should be specific to the compiler, i.e. the Xcode version that the runner uses. macOS 12 is two years old too, so my best guess would be that it doesn't support bit_cast either yet.

Reply:
https://developer.apple.com/xcode/cpp/ says bit-casting requires XCode 14.3. Seems like that's only available on MacOS 13...

Reply:
Ah okay, thanks for checking. I think as long as we're building against an SDK older than that (in CI), we are going to get an error, regardless of whether we use a newer runner. I'll update the comment.

Reply:
That's a neat site btw, have to bookmark that.

Reply:
Even better: https://en.cppreference.com/w/cpp/compiler_support

Apparently they also track apple clang now, which they previously haven't which is why I defaulted to the apple site.